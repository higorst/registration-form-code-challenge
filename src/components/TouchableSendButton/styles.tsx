import styled from 'styled-components/native'
import { TouchableOpacityProps, TextProps, Dimensions } from 'react-native'
import { Colors } from '../../styles/Colors'

interface ButtonType extends TouchableOpacityProps {
    color?: string
    sending?: boolean
}
interface TitleButtonType extends TextProps {
    color?: string
    fontSize?: number
}

export const TitleButton = styled.Text`
    color: #ffffff;
    text-align: center;  
    font-weight: bold;
    font-size: ${(props: TitleButtonType) => `${props.fontSize}px` || `20px`};
`

export const Touchable = styled.TouchableOpacity`
    padding: 15px;
    width: ${`${Dimensions.get('window').width * 0.9}px`};

    align-self: center;
    align-content: center;
    justify-content: center;

    border-radius: 10px; 
    flex-direction: row;

    justify-content: space-evenly;
    align-items: center;

    border-width: 2px;

    z-index: 5;
    border-color: ${(props: ButtonType) => 
        props.sending ? Colors.white : Colors.backgroundApp};
    background-color: ${(props: ButtonType) => 
        props.sending ? Colors.backgroundApp : 
        props.color ? props.color : Colors.primary};
`