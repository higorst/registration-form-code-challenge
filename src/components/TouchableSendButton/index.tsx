import React from 'react'
import { ActivityIndicator } from 'react-native'
import { Constants } from '../../constants';
import { Colors } from '../../styles/Colors';
import { Touchable, TitleButton } from './styles'

interface PropsButton {
    color: string
    title: string

    // flags
    sending?: boolean
    onPress?: any
}
function TouchableSendButton(props: PropsButton) {

    return (
        <Touchable
            color={props.color}
            sending={props.sending}
            onPress={props.onPress}
        >
            <TitleButton
                fontSize={Constants.fontSizeTitleTouchable}
            >{props.sending ? Constants.titleButtonSending : props.title}</TitleButton>
            {props.sending && <ActivityIndicator size="small" color={Colors.white} />}

        </Touchable>
    )
}

export default TouchableSendButton