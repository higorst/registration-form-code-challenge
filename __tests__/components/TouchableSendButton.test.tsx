import React from 'react';
import renderer from 'react-test-renderer';
import TouchableSendButton from '../../src/components/TouchableSendButton/index'

jest.useFakeTimers()
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper')

describe('<TouchableSendButton />', () => {
    it('renders TouchableSendButton Component correctly', () => {
        const tree = renderer.create(
            <TouchableSendButton
                title='Enviado Snapshot'
                color='#013893'
                onPress={() => {}}
                sending={false}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});