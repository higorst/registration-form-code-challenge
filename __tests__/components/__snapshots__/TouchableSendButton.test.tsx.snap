// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TouchableSendButton /> renders TouchableSendButton Component correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#013893",
      "borderColor": "#4e61c4",
      "borderRadius": 10,
      "borderWidth": 2,
      "flexDirection": "row",
      "justifyContent": "space-evenly",
      "opacity": 1,
      "paddingBottom": 15,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 15,
      "width": 675,
      "zIndex": 5,
    }
  }
>
  <Text
    fontSize={18}
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "center",
        },
      ]
    }
  >
    Enviado Snapshot
  </Text>
</View>
`;
